src/MongoDB.Driver/Encryption/LibMongoCryptControllerBase.cs

```diff
@@ -170,6 +172,7 @@ private void ParseKmsEndPoint(string value, out string host, out int port)
                 host = value;
                 port = 443;
             }
+            dnsEndPoint = new DnsEndPoint(host, port);
```

Comment:
I will double-check this line, not sure whether `DnsEndpoint` can be used for the above `else` branch